java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example57_KernelHooks.java

```diff
@@ -365,4 +369,51 @@ private static void chatCompletionHook(Kernel kernel) {
         }
     }
 
+    /**
+     * Show use of hooks added on a single invocations.
+     *
+     * @param kernel
+     */
+    private static void invocationHook(Kernel kernel) {
+        // Initialize prompt
+        String functionPrompt = "Write a paragraph about hats";
+
+        var writerFunction = KernelFunctionFromPrompt.builder()
+            .withTemplate(functionPrompt)
+            .withName("Writer")
+            .withDefaultExecutionSettings(PromptExecutionSettings
+                .builder()
+                .withMaxTokens(1000)
+                .withTemperature(1)
+                .withTopP(0.5)
+                .build())
+            .build();
+
+        HookService hooks = new HookService();
```

Comment:
Consider renaming this from "service" since that is an overloaded term in SK. Maybe InvocationHooks or just Hooks.